projects/rero/ng-core/src/lib/record/record.service.ts

```diff
@@ -360,48 +363,19 @@ export class RecordService {
     field: string,
     q: string
   ): Observable<Array<string>> {
-    return this._http.get<Array<string>>(`${this._apiService.baseUrl}${url}`, {
+    return this.http.get<Array<string>>(`${this.apiService.baseUrl}${url}`, {
       params: { resource, field, q },
     });
   }
 
   /**
    * Error handling during api call process.
    * @param error - HttpErrorResponse
+   * @param resourceName- Name of current resource
    * @return throwError
    */
-  private _handleError(error: HttpErrorResponse): Observable<Error> {
-
-    // handle Marshmallow errors
-    //   Python marshmallow library can raise ValidationError. In this case, the message is simply 'Validation error'
-    //   but list of errors are passed into the `errors` fields.
-    if (error.error?.errors) {
-      const message = error.error.errors
-          .map(err => this._translateService.instant(err.message))
-          .join(' ; ');
-      return throwError({ status: error.status, title: message});
-    }
-
-    // check if we have possible custom error message to display
-    if (error.status === 400 && error.error.hasOwnProperty('message')) {
-      let message = error.error.message;
-      message = message.replace(/^Validation error: /, '').trim();  // Remove Invenio `ValidationError` header
-      return throwError({ status: error.status, title: this._translateService.instant(message) });
-    }
-
-    // If not, then return default message
-    switch (error.status) {
-      case 400:
-        return throwError({ status: error.status, title: this._translateService.instant('Bad request') });
-      case 401:
-        return throwError({ status: error.status, title: this._translateService.instant('Unauthorized') });
-      case 403:
-        return throwError({ status: error.status, title: this._translateService.instant('Forbidden') });
-      case 404:
-        return throwError({ status: error.status, title: this._translateService.instant('Not found') });
-      default:
-        return throwError({ status: 500, title: this._translateService.instant('An error occurred') });
-    }
+  private _handleError(error: HttpErrorResponse, resourceName?: string): Observable<never> {
```

Comment:
`resourceName` can be removed.

Reply:
This parameter is used in the UI. It's to have the same signature.